### PR TITLE
Mechs now use equipment flags to check if they can attach

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Add savannah flag
Change checks to now check for flags instead of types

## Why It's Good For The Game
When you uhm have less shitcode uhm

## Changelog

:cl:
refactor: changed how whether a mech equipment is attachable is determined, please report if any previously attachable equipments are bugged
/:cl:

### CODE REVIEW

**Comment by zxaber** [comment]
Instead of giving each mech type its own flag, why not make MECH_COMBAT, MECH_CONSTRUCTION, MECH_MEDICAL, and then give mechs the ones that apply to that mech?

Also, how does this all function with the hidden weapons bay traitor item?

**Comment by TiviPlus** [comment]
> Instead of giving each mech type its own flag, why not make MECH_COMBAT, MECH_CONSTRUCTION, MECH_MEDICAL, and then give mechs the ones that apply to that mech?

it is for the Exosuit UI
> Also, how does this all function with the hidden weapons bay traitor item?

At least temp removed during equipment rework im still think about how best to readd it

File: code/modules/vehicles/mecha/equipment/mecha_equipment.dm [hunk 1]
```diff
@@ -21,7 +21,7 @@
 	///Bitflag. Determines the range of the equipment.
 	var/range = MECHA_MELEE
 	/// Bitflag. Used by exosuit fabricator to assign sub-categories based on which exosuits can equip this.
-	var/mech_flags = NONE
+	var/mech_flags = EXOSUIT_MODULE_ALL
```
Comment (timothymtorres) [inline code/modules/vehicles/mecha/equipment/mecha_equipment.dm:24]: It's probably a good idea to just use the `ALL` define. That way people in the future don't have to keep adding their flags to the `EXOSUIT_MODULE_ALL` define.

Comment (timothymtorres) [reply code/modules/vehicles/mecha/equipment/mecha_equipment.dm:24]:
```suggestion
	var/mech_flags = ALL
```

File: code/__DEFINES/exosuit_fab.dm [hunk 2]
```diff
@@ -23,13 +23,17 @@
...
 
 /// Module is compatible with "Working" Exosuit models - Ripley and Clarke
 #define EXOSUIT_MODULE_WORKING EXOSUIT_MODULE_RIPLEY | EXOSUIT_MODULE_CLARKE
 /// Module is compatible with "Combat" Exosuit models - Gygax, H.O.N.K, Durand and Phazon
-#define EXOSUIT_MODULE_COMBAT EXOSUIT_MODULE_GYGAX | EXOSUIT_MODULE_HONK | EXOSUIT_MODULE_DURAND | EXOSUIT_MODULE_PHAZON
+#define EXOSUIT_MODULE_COMBAT EXOSUIT_MODULE_GYGAX | EXOSUIT_MODULE_HONK | EXOSUIT_MODULE_DURAND | EXOSUIT_MODULE_PHAZON | EXOSUIT_MODULE_SAVANNAH
 /// Module is compatible with "Medical" Exosuit modelsm - Odysseus
 #define EXOSUIT_MODULE_MEDICAL EXOSUIT_MODULE_ODYSSEUS
+/// Module is compatible with all mechs
+#define EXOSUIT_MODULE_ALL EXOSUIT_MODULE_WORKING | EXOSUIT_MODULE_COMBAT | EXOSUIT_MODULE_MEDICAL
```
Comment (timothymtorres) [inline code/__DEFINES/exosuit_fab.dm:36]: This isn't needed.  Just use the `ALL` define when you need to set all the flags.

```suggestion
```

**Pull request merged**